h/static/scripts/directive/h-autofocus.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+/** An attribute directive that focuses an <input> when it is shown.
```

Comment:
Need to re-word this slightly to clarify when it is triggered. s/shown/"added to the DOM".

If the `<input>` was shown with `ng-show`, the current implementation would not work. The current implementation is nice and simple so rather than changing it to work with `ng-show`, I'd just re-word this sentence slightly.